### ISSUE TITLE
External Search not loading consistently

### ISSUE THREAD

**Issue by MargaretA-UO** [uoregon-libraries/uolib-primo-nui#44]
We use a script that is apparently only loading the external search (Worldcat and Google Scholar) if other elements of the page have loaded at the point that the external search code has been told to run.  Tamara at the Alliance writes:
In the PCSG's central package version of the module, the script attempts to create the facet every 100 milliseconds until it succeeds: https://github.com/alliance-pcsg/primo-explore-central-package/blob/71dd6d3935e48dec7ce8a983e67ddd83a11cba4d/CENTRAL_PACKAGE/js/custom.js#L766

In UO's version, which overwrites the central package module starting in line 339 of your custom.js, the directive fires only once. If the required services weren't loaded yet, the facet won't appear.

If you want to keep UO's unique version, you can modify the "link" function within the module to wrap the facet creation in an interval like PCSG's. The code below is theoretical, not tested, but should loop every 100 ms until a facet in position 3 exists, then add External Search:

var checkFacet = setInterval(function () {
  if (prmFacetCtrl.facets[facetPos]) { 
    if (prmFacetCtrl.facets[facetPos].name !== facetTitle) {
      prmFacetCtrl.facets.splice(facetPos, 0, {
        name: facetTitle,
        displayedType: 'exact',
        limitCount: 0,
        facetGroupCollapsed: false,
        values: [],
      })
    }
    clearInterval(checkFacet) 
  }
}, 100);

**Comment by MargaretA-UO** [comment]
Note:  this is not currently working in either Primo or VE

**Comment by jechols** [comment]
Looks to be related to issues I had with #50

**Comment by MargaretA-UO** [comment]
This is core functionality that we should do our best to get working consistently.

**Comment by jechols** [comment]
That's unlikely to be helpful. It looks almost exactly like what we're already doing. We didn't invent the code ourselves; we simply made use of something Corey found from the alliance. The problem isn't the code, it's how it interacts with the facet collapse somehow. At least, that was my latest test results - kill facet collapse and the external bit seems to be a whole lot more consistent.

Given that this has been a problem for years, and I've already spent several hours trying to make this work with the facet collapsing, this probably won't get any attention for a while. If there are priorities of what needs to happen vs. what would be nice to have, that can direct my work better, but for now I'm just blindly trying to get as many "things" working as possible, which by necessity means dropping individual items that are costing too much time and/or are close enough to current functionality. This issue is both of those things.

**Comment by jechols** [comment]
My assessments continue to be incorrect. Taking the collapsable facets out of the equation now results in the external facet *never* working.

Given how intermittent this problem is, it's safe to look at all "conclusions" with a fair bit of skepticism.

**Comment by MargaretA-UO** [comment]
Ok, new approach - I think I can now configure the collapsing of the facets myself, in Alma.  I've set it the way I think this should work to have the first three facets show 5 links, and the rest collapsed. 
So if you want to experiment again with taking out the code for collapsing those links, and see if it's working without that, then you can drop that piece of the mess on your side, and focus back on the External Results piece.

**Comment by jechols** [comment]
This is now complete!!

**Comment by MargaretA-UO** [comment]
Thank you!
